src/NServiceBus.Core/Pipeline/Incoming/ITransportReceiveContext.cs

```diff
@@ -11,5 +12,11 @@ public interface ITransportReceiveContext : IBehaviorContext
         /// The physical message being processed.
         /// </summary>
         IncomingMessage Message { get; }
+    
+        /// <summary>
+        /// Allows the pipeline to flag that it has been aborted and the receive operation should be rolled back. 
+        /// It also allows the transport to communicate to the pipeline to abort if possible.
+        /// </summary>
+        CancellationTokenSource CancellationTokenSource { get; }
```

Comment:
I'll go with the new API and adjust the comments to not talk about bidirectional communication for now. We can handle that later.